apps/daimo-bot/api/PaymentActionProcessor.ts

```diff
@@ -0,0 +1,235 @@
+import { trpcClient } from "./trpcClient";
+import { SendCastOptions, TRPCClient, WebhookEvent } from "./types";
+
+dotenv.config();
+
+assert(!!process.env.NEYNAR_API_KEY, "NEYNAR_API_KEY is not defined");
+const _neynarClient = new NeynarAPIClient(process.env.NEYNAR_API_KEY);
+
+assert(
+  !!process.env.DAIMOBOT_SIGNER_UUID,
+  "DAIMOBOT_SIGNER_UUID is not defined"
+);
+const DAIMOBOT_SIGNER_UUID = process.env.DAIMOBOT_SIGNER_UUID;
+
+export class PaymentActionProcessor {
+  private text: string;
+  private castId: string;
+  private senderFid: number;
+  private authorUsername: string;
+  private parentAuthorFid: number | null;
+  private trpcClient: TRPCClient;
+  private neynarClient: NeynarAPIClient;
+
+  constructor(
+    event: WebhookEvent,
+    trpc: TRPCClient = trpcClient,
+    neynarClient: NeynarAPIClient = _neynarClient
+  ) {
+    const { data } = event;
+    this.text = data.text;
+    this.castId = data.hash;
+    this.authorUsername = data.author.username;
+    this.parentAuthorFid = data.parent_author.fid;
+    this.senderFid = data.author.fid;
+
+    this.trpcClient = trpc;
+    this.neynarClient = neynarClient;
+  }
+
+  async process() {
+    // Processes the incoming webhook event from Farcaster, extracting either a "request" or "pay" command,
+    // and performs actions based on the command type.
+
+    // 4 cases of Payment request:
+
+    // "@daimobot request":
+    // Case 1: Alice doesn't have FC linked ❌, requests $ from anyone (open-ended post)
+    //      Action 1: Daimobot responds with a link to register with Farcaster. Alice registers, then Daimobot responds with a link to request $
+    // Case 2: Alice has FC linked ✅, requests $ from anyone
+    //      Action 2: Daimobot responses with link that requests $ from anyone to Alice's Daimo address
+
+    // "@daimobot pay":
+    // Case 3: Alice responds to Bobs post to pay him, Bob doesn't have FC linked ❌
+    //     Action 3:  Daimobot responds with a link to register with Farcaster. Bob registers, then Daimobot responds with a link to request $
+    // Case 4: Alice responds to Bobs post to pay him, Bob has FC linked ✅
+    //     Action 4: Daimobot responds with link that requests $ from anyone to Bob's Daimo address
+
+    const daimobotCommand = this._tryExtractCommand();
+    if (!daimobotCommand) {
+      console.log("Cast follows neither request nor pay format.");
+      this.publishCastReply(
+        "Hi! To use me, please use the format `@daimobot request $<amount>` or `@daimobot pay $<amount>`."
+      );
+      return;
+    }
+
+    const { action, cleanedAmount } = daimobotCommand;
+    switch (action) {
+      case "request": {
+        // See if sender has Farcaster linked
+        console.log(
+          `[DAIMOBOT REQUEST] lookupEthereumAccountByFid for FID: ${this.senderFid}`
+        );
+        const senderEthAccount =
+          await this.trpcClient.lookupEthereumAccountByFid.query({
+            fid: this.senderFid,
+          });
```

Comment:
i think we could factor out a nice feature here along lines of:

```
const sender = await this.lookupFid(this.senderFid);
```
...where lookupFid returns fid + username + (in future, connected EAccount, if any)

Reply:
Did you mean within this DaimobotProcessor scope or via trpcClient?

If you meant the former, done in PR fixes. If you meant the latter, can easily tack on in follow-on issue